bluepyopt/ephys/parameterscalers.py

```diff
@@ -196,3 +198,124 @@ def __str__(self):
         """String representation"""
 
         return self.distribution
+
+
+class NrnSegmentSectionDistanceScaler(ParameterScaler, DictMixin):
+
+    """Scaler based on distance from soma"""
+
+    SERIALIZED_FIELDS = (
+        "name",
+        "comment",
+        "distribution",
+        "dist_param_names",
+        "ref_sec",
+        "ref_location",
+    )
+
+    def __init__(
+        self,
+        name=None,
+        distribution=None,
+        comment="",
+        dist_param_names=None,
+        ref_sec="soma[0]",
+        ref_location=0,
+    ):
+        """Constructor
+
+        Args:
+            name (str): name of this object
+            distribution (str): distribution of parameter dependent on distance
+                from soma. string can contain `distance` and/or `value` as
+                placeholders for the distance to the soma and parameter value
+                respectivily
+            dist_param_names (list): list of names of parameters that parametrise
+                the distribution. These names will become attributes of this
+                object.
+                The distribution string should contain these names, and they
+                will be replaced by values of the corresponding attributes
+            ref_sec (str): string with name of reference section to compute distance (e.g. "soma[0]", "dend[2]")
+            ref_location (float): location along the soma used as origin
+                from which to compute the distances. Expressed as a fraction
+                (between 0.0 and 1.0).
+        """
+
+        super(NrnSegmentSectionDistanceScaler, self).__init__(name, comment)
+        self.distribution = distribution
+
+        self.dist_param_names = dist_param_names
+        self.ref_location = ref_location
+        self.ref_sec = ref_sec
+
+        if not (0.0 <= self.ref_location <= 1.0):
+            raise ValueError("ref_location must be between 0 and 1.")
+
+        if self.dist_param_names is not None:
+            for dist_param_name in self.dist_param_names:
+                if dist_param_name not in self.distribution:
+                    raise ValueError(
+                        'NrnSegmentSomaDistanceScaler: "{%s}" '
```

Comment:
```suggestion
                        'NrnSegmentSectionDistanceScaler: "{%s}" '
```
@darshanmandge  With the +- button, like this!

Reply:
Thanks. This feature is useful to know.